src/linux/network.rs

```diff
@@ -76,12 +76,30 @@ impl NetworksExt for Networks {
     fn refresh_networks_list(&mut self) {
         if let Ok(dir) = std::fs::read_dir("/sys/class/net/") {
             let mut data = vec![0; 30];
-            for entry in dir.flatten() {
+
+            let entries: Vec<DirEntry> = dir
+                .flatten()
+                .filter(|e| e.file_name().into_string().is_ok())
+                .collect();
+            let names: Vec<String> = entries
+                .iter()
+                .filter_map(|e| e.file_name().into_string().ok())
+                .collect();
+
+            // Remove interfaces that are gone
+            for name in self
+                .interfaces
+                .keys()
+                .filter(|n| !names.contains(n))
+                .map(|n| n.to_owned())
+                .collect::<Vec<String>>()
+            {
+                self.interfaces.remove(&name);
+            }
```

Comment:
Instead of doing that, why not doing something like this:

```suggestion
            self.interfaces = self
                .interfaces
                .iter()
                .filter(|n| names.contains(n))
                .collect::<Vec<_>>();
```

?

Reply:
self.interfaces is a HashMap

Reply:
I updated my example. I was basing it on https://doc.rust-lang.org/nightly/std/collections/struct.HashMap.html#impl-FromIterator%3C(K%2C%20V)%3E

Reply:
It's definitely simpler and more readable, but isn't it less efficient since we would be rebuilding the hashmap at every refresh, including when no interface was removed?

Reply:
It's a good question actually. Running benchmarks would allow us to know I guess.